### PR TITLE
Async dashboard

### PR DESCRIPTION
**Pull request by dennisreimann**
Builds on #3907 and makes all the tiles async.

Also adds a weekly/monthly period selector for the app sales.

### CODE REVIEW

**Comment by dennisreimann** [comment]
@pavlenex In #3907 we said to keep them separate, because it includes amny changes already. I'll rebase this one once #3907 is merged - wanted to bring this one up so that dstrukt and you can see the async dashboard already.

**Comment by pavlenex** [comment]
Just user-tested, seems really nice.
Screenshot 2022-06-29 at 20 36 26
Screenshot 2022-06-29 at 20 36 33

**Review by dstrukt — APPROVED** [review]
Finally got a moment to do some proper testing.

LGTM!

Tis a nice change.

File: BTCPayServer/Controllers/UIStoresController.Dashboard.cs [hunk 1]
```diff
@@ -39,36 +30,64 @@ public async Task<IActionResult> Dashboard()
...
-                        return appData;
-                    })
-                    .ToList();
-            }
+            if (!vm.WalletEnabled && !vm.LightningEnabled) return View(vm);
             
+            var userId = GetUserId();
+            var apps = await _appService.GetAllApps(userId, false, store.Id);
+            vm.Apps = apps
+                .Select(a => _appService.GetAppDataIfOwner(userId, a.Id).Result)
```
Comment (NicolasDorier) [inline BTCPayServer/Controllers/UIStoresController.Dashboard.cs:43]: Never ever call `.Result`. This inevitably create deadlocks of the whole app at random time. Need to do a classic loop here.

Comment (dennisreimann) [reply BTCPayServer/Controllers/UIStoresController.Dashboard.cs]: Thanks, fixed in [5d58370](https://github.com/btcpayserver/btcpayserver/pull/3916/commits/5d58370aa1e9fc87651533fdc761e768eaaf0f7e).

**Pull request merged**